src/types/itemTypes.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+// Item structure coming from the Search Results API response
+export interface SearchResultsItem {
+  "@id": string
+  idNyplSourceId?: ItemSourceID
+  accessMessage?: ItemAccessMessage[]
+  shelfMark?: string[]
+  electronicLocator?: string[]
+  status?: ItemStatus[]
+  enumerationChronology?: string[]
+  formatLiteral?: string[]
+  idBarcode?: string[]
+  holdingLocation?: ItemLocation[]
+  aeonUrl?: string[]
+  physRequestable?: boolean
+  eddRequestable?: boolean
+}
+
+export interface ItemAccessMessage {
+  "@id": string
+  prefLabel?: string
+}
+
+export interface ItemSourceID {
+  "@type": string
+  "@value": string
+}
+
+export interface ItemStatus {
+  "@id": string
+  prefLabel?: string
```

Comment:
This pair
```
"@id": string
  prefLabel?: string`
```

appears a few times. I'm not sure if there's a better interface name for this that can be more generic. Both `ItemAccessMessage` and `ItemStatus` are the same, so it can be confusing when implementing them whether there's a difference or not.

Reply:
I think it would be worthwhile to have a generic `JSONLDValue` type.